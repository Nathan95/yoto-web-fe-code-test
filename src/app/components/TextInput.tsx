import React from 'react';

export interface TextInputTypes {
  placeholder: string;
  datatestid?: string;
  onChange: (args: any) => void;
  value: string;
  className?: string;
}

export const TextInput  = ({
  placeholder,
  datatestid,
  onChange,
  value,
  className,
}: TextInputTypes) => {
  return (
    <input
      name="input"
      type="text"
      placeholder={placeholder}
      data-testid={datatestid}
      onChange={onChange}
      value={value}
      className={className}
    />
  );
};

export default TextInput;
