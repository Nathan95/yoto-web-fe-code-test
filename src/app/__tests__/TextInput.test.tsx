import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import TextInput from '../components/TextInput';

const mock = {
  placholder: 'text',
  onChange: () => console.log('hello'),
  value: 'value',
};

describe('InputField', () => {
  const props = mock;
  it('renders the component', () => {
    expect(
      <TextInput
        placeholder="text"
        onChange={() => console.log('hello')}
        value=""
      />,
    ).toMatchSnapshot();
  });
});
