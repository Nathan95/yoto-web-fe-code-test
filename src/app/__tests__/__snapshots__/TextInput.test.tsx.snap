// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders the component 1`] = `
<TextInput
  onChange={[Function]}
  placeholder="text"
  value=""
/>
`;
